astropy/stats/biweight.py

```diff
@@ -328,51 +347,59 @@ def biweight_midvariance(data, c=9.0, M=None, axis=None,
     0.97362869104
     """
 
+    if ignore_nan:
+        median_func = np.nanmedian
+        sum_func = np.nansum
+    else:
+        median_func = np.median
+        sum_func = np.sum
+
     data = np.asanyarray(data).astype(np.float64)
 
     if M is None:
-        M = np.median(data, axis=axis)
+        M = median_func(data, axis=axis)
     if axis is not None:
         M = _expand_dims(M, axis=axis)  # NUMPY_LT_1_18
 
     # set up the differences
     d = data - M
 
     # set up the weighting
-    mad = median_absolute_deviation(data, axis=axis)
+    mad = median_absolute_deviation(data, axis=axis, ignore_nan=ignore_nan)
 
     if axis is None and mad == 0.:
         return 0.  # return zero if data is a constant array
 
     if axis is not None:
         mad = _expand_dims(mad, axis=axis)  # NUMPY_LT_1_18
-        const_mask = (mad == 0.)
-        mad[const_mask] = 1.  # prevent divide by zero
+        mad[mad == 0] = 1.  # prevent divide by zero
 
     u = d / (c * mad)
 
     # now remove the outlier points
-    mask = np.abs(u) < 1
+    # ignore RuntimeWarnings for comparisons with NaN data values
+    with np.errstate(invalid='ignore'):
+        mask = np.abs(u) < 1
     u = u ** 2
 
+    nanmask = np.ones(data.shape)
```

Comment:
nitpicking, but this var name is a bit confusing as otherwise we use "mask" the other way around.

Reply:
Yeah I know, but then I would have to invert the mask to use it.  This prevents an unnecessary computation.  I'm working on a followup PR to allowed masked arrays that will change this line.  Name suggestions for an inverse mask?

Reply:
`inversed_mask`? :sweat_smile:

Reply:
`inverted_mask`? `son_of_the_mask`?

Reply:
yes, I don't suggest to invert it, it was more like a naming issue. But then again, the code after that is trivial, it's not one of those cases where the variable was used again hundreds of  lines later.

Reply:
I know you're talking about cone search, Brigitta! :joy_cat:

Reply:
I wasn't even, honestly I haven't looked into that code at all, but now I know to stay away 😺